Unify user session lifetime configuration

michalmisiewicz opened the pull request:
In current version of Airflow user session lifetime can be configured by `session_lifetime_days` and `force_log_out_after` options. In practise only `session_lifetime_days` has impact on session lifetime, but it is limited to values in day. In this PR I have simplified user session lifetime configuration.

michalmisiewicz committed:
Add datumo.io to Airflow family (#11692)

mik-laj committed:
Fix example DAGs in pip packages (#11687)

potix2 committed:
Add D200 pydocstyle check (#11688)

ryanahamilton committed:
Improve legibility with greater contrast ratio in footer and pagination (#11690)

FloChehab committed:
fix pod launcher rolebinding in helm chart (#11675)

* Followup to #11034
* Was not referencing the correct kind of resources if multiNamespaceMode = False

mik-laj committed:
Extract Kubernetes command to separate file (#11669)

* Move Kubernetes command to seperate file

* fixup! Move Kubernetes command to seperate file

nheer committed:
Fix documentation for PythonVirtualenvOperator (#11700)

Fixed the op_args type description

msumit committed:
Update TIs with a proper lock (#11683)

turbaszek committed:
Revert "Refactor celery worker command (#11336)" (#11698)

This reverts commit 02ce45cafec22a0a80257b2144d99ec8bb41c961.
That refactored Clery worker to be compatible with 5.0. However this
introduced some incompatibilities.

Closes: #11622
Closes: #11697

msumit committed:
Make table header text nowrap (#11689)

mik-laj committed:
Consistent use images in Helm Chart (#11701)

mik-laj committed:
Better file extension for Helm template (#11702)

* Better file extension for Helm template

* fixup! Better file extension for Helm template

mik-laj committed:
Enforce strict rules for yamllint (#11709)

ashb committed:
Fix doc errors in google provider files. (#11713)

These files aren't _currently_ rendered/parsed by autoapi, but I was
exploring making them parseable and ran in to some sphinx formatting
errors.

The `Args:` change is because pydocstyle thinks that is a special word, but
we don't want it to be.

jbampton committed:
Fix case of GitHub (#11398)

jmcarp committed:
Check response status in slack webhook hook. (#11620)

mik-laj committed:
Remove unused value in Helm Chart - podMutation (#11703)

potiuk committed:
Also cancel duplicated Build Image runs for master pushes (#11655)

We removed the "skipEventType" for CI jobs but we still had
the skipping for Build Image and killing builds with failed jobs.

This further optimises our job usage.

jbampton committed:
Fix case of JavaScript (#11718)

Changes were:
- `javascript` to `JavaScript`

kosteev committed:
Retry requests in case of error in Google ML Engine Hook (#11712)

joebeeson committed:
Add support for setting ciphers for SFTPHook (#11720)

ryanahamilton committed:
Refactor "loading" state of DAGs view to remove visual jank (#11725)

ashb committed:
Fix formatting errors introduced in #11720 (#11733)

potix2 committed:
Add type hints to  aws provider (#11531)

* Added type hints to aws provider

* Update airflow/providers/amazon/aws/log/s3_task_handler.py

* Fix expectation for submit_job

* Fix documentation

Co-authored-by: Kamil Breguła <mik-laj@users.noreply.github.com>

tanjinP committed:
Add Google Cloud Memorystore Memcached Operators (#10121)

Co-authored-by: Tobiasz Kędzierski <tobiasz.kedzierski@polidea.com>
Co-authored-by: Kamil Breguła <mik-laj@users.noreply.github.com>

mik-laj committed:
Fix static checks after merging #10121 (#11737)

potiuk committed:
Fixes ROVIDERS -> PROVIDERS typo in Dockerfile (#11738)

There was a typo in the original file when review was made in
the #11529 but apparently this typo was still left in one place
and as the result, providers have not been installed in the
master Dockerfile.

Fixes #11695

mjpieters committed:
Ensure task logs go to the correct try number file (#11723)

The run context (logging context) accesses task instance attributes via the log_filename_template configuration.

Fixes #11717.

michalmisiewicz committed:
Unify user session lifetime configuration

michalmisiewicz committed:
change session_lifetime_minutes option type to int

michalmisiewicz committed:
improve session_lifetime_minutes option description

ashb committed:
Stop scheduler from thinking that upstream_failed tasks are running (#11730)

This was messing up the "max_active_runs" calculation, and this fix is a
"hack" until we add a better approach of adding a queued state to
DagRuns -- at which point we don't even have to do this calculation at
all.

kaxil committed:
Use unittest.mock instead of backported mock library (#11643)

mock is now part of the Python standard library, available as unittest.mock in Python 3.3 onwards.

mik-laj committed:
Improve Cloud Memorystore for Redis example (#11735)

yuqian90 committed:
Speed up `dag.clear()` when clearing lots of ExternalTaskSensor and  ExternalTaskMarker (#11184)

This is an improvement to the UI response time when clearing dozens of DagRuns of large DAGs (thousands of tasks) containing many ExternalTaskSensor + ExternalTaskMarker pairs. In the current implementation, clearing tasks can get slow especially if the user chooses to clear with Future, Downstream and Recursive all selected.

This PR speeds it up. There are two major improvements:

Updating self._task_group in dag.sub_dag() is improved to not deep copy _task_group because it's a waste of time. Instead, do something like dag.task_dict, set it to None first and then copy explicitly.
Pass the TaskInstance already visited down the recursive calls of dag.clear() as visited_external_tis. This speeds up the example in test_clear_overlapping_external_task_marker by almost five folds.
For real large dags containing 500 tasks set up in a similar manner, the time it takes to clear 30 DagRun is cut from around 100s to less than 10s.

shivanshs9 committed:
Docs: Typo in Smart sensor (#11746)

``sensors_enabled`` config key was mistyped.

zacharya19 commented on tests/www/test_app.py:
```diff
@@ -215,3 +217,27 @@ def test_should_set_sqlalchemy_engine_options(self):
+        app_path = str(Path(__file__).parents[2].joinpath('airflow', 'www', 'app.py'))
+        warns = [w for w in warns if w.filename == app_path]
+        self.assertEqual(len(warns), 2)
+        self.assertEqual(warns[0].message.args[0],
```
My main issue with this function is that if we tomorrow add new deprecation warnings it can break this test, while it shouldn't.
Why not just check in general that those messages exists in warns? Why filter by filename and be specific on the index number?

michalmisiewicz replied:
Resolved

michalmisiewicz committed:
modify should_emit_deprecation_warnings test

kaxil committed:
Fix Static Checks (#11749)

zacharya19 reviewed (approved):
LGTM

tooptoop4 commented on airflow/www/app.py:
```diff
@@ -66,8 +66,16 @@ def create_app(config=None, testing=False, app_name="Airflow"):
+                      DeprecationWarning)
+
+    if conf.has_option('webserver', 'FORCE_LOG_OUT_AFTER'):
+        warnings.warn('FORCE_LOG_OUT_AFTER option is deprecated. Please use `SESSION_LIFETIME_MINUTES`',
```
not deprecated, removed. can remove this and just populate the UPDATING.MD

michalmisiewicz replied:
I believe that we should warn user about removed options, as this is breaking change. Warnings can be removed in next Airflow release.  I can changed message to `option is removed. Please use ...`
Please let me know if I should remove this warning.

tooptoop4 replied:
option is removed. Please use ...

michalmisiewicz committed:
change deprecation message

mik-laj commented on airflow/www/app.py:
```diff
@@ -66,8 +66,16 @@ def create_app(config=None, testing=False, app_name="Airflow"):
 
-    session_lifetime_days = conf.getint('webserver', 'SESSION_LIFETIME_DAYS', fallback=30)
-    flask_app.config['PERMANENT_SESSION_LIFETIME'] = timedelta(days=session_lifetime_days)
+    if conf.has_option('webserver', 'SESSION_LIFETIME_DAYS'):
```
I think we should not warn users about configuration options that do not work if the user has the correct configuration options for Airflow 2.0. This will allow us to use one configuration file for Airflow 1.10 and 2.0.  This is essential if users want to use Helm Chart and other generic deployment tools.

So I propose that you check if there is a `SESSION_LIFETIME_MINUTES` option in the config file. If so, use it.
If it is not there, we should display a deprecation warning and we should try to maintain backward compatibility.  In this case, this means that the warning should display a message that option `SESSION_LIFETIME_DAYS` has been renamed to `SESSION_LIFETIME_MINUTES` and the unit from day to hour and that option `FORCE_LOG_OUT_AFTER` has been removed.

When this change has been merged, please create a ticket requesting a new rule for [airflow upgrade-check](https://github.com/apache/airflow/issues/8765).

michalmisiewicz replied:
@mik-laj I've updated the code

michalmisiewicz replied:
@mik-laj are we ready to go ?

michalmisiewicz committed:
requested changes

michalmisiewicz committed:
test if PERMANENT_SESSION_LIFETIME is set when SESSION_LIFETIME_MINUTES is not provided

potiuk committed:
Enables splitting tests into smaller chunks (#11659)

We've implemented the capability of running the tests in smaller
chunks and selective running only some of those, but this
capability have been disabled by mistake by default setting of
TEST_TYPE to "All" and not removing it when TEST_TYPES are set
to the sets of tests that should be run.

This should speed up many of our tests and also hopefully
lower the chance of EXIT 137 errors.

mik-laj committed:
Improve web server stopping (#11734)

darwinyip committed:
Add reattach flag to ECSOperator (#10643)

..so that whenever the Airflow server restarts, it does not leave rogue ECS Tasks. Instead the operator will seek for any running instance and attach to it.

ashb committed:
Ensure that manually creating a DAG run doesn't "block" the scheduler (#11732)

It was possible to "block" the scheduler such that it would not
schedule or queue tasks for a dag if you triggered a DAG run when the
DAG was already at the max active runs.

This approach works around the problem for now, but a better longer term
fix for this would be to introduce a "queued" state for DagRuns, and
then when manually creating dag runs (or clearing) set it to queued, and
only have the scheduler set DagRuns to running, nothing else -- this
would mean we wouldn't need to examine active runs in the TI part of the
scheduler loop, only in DagRun creation part.

Fixes #11582

ashb committed:
Fix FAB actions with for models with composite PKs (#11753)

This fixes #11513 -- and has been submitted upstream to FAB as
https://github.com/dpgaspar/Flask-AppBuilder/pull/1493, once that is
merged we can remove this override.

potiuk committed:
Add test types displayed in CI (#11770)

This is a follow up after #11659. Thanks to this one, it will
be immediately visible in CI which types of tests are run.

This change has been lost during one of the rebases, so it is
brought back now.

ryanahamilton committed:
fix: Override FAB styling (#11752)

yangrong688 committed:
Add uSmart Securities to the INTHEWILD.md (#11757)

potiuk committed:
The .tar.gz provider packages are installable now. (#11630)

The packages lacked setup.py and they could not be installed.

This change automatically generates setup.py for the packages and
adds them to the packages.

Fixes: #11546

kaxil committed:
Add reference link for KubernetesPodOperator in kubernetes.rst (#11782)

This makes it easy to go to the class definition and find the arguments/params that can be passed to the Operator

potiuk committed:
The .pypirc file is read from docker-context-files (#11779)

If you used context from git repo, the .piprc file was missing and
COPY in Dockerfile is not conditional.

This change copies the .pypirc conditionally from the
docker-context-files folder instead.

Also it was needlessly copied in the main image where it is not
needed and it was even dangerous to do so.

ryanahamilton committed:
Improve the visual presentation of the variable import form (#11783)

No functional changes, just visual enhancement of the variable import
form. This simply refactors the markup slightly to better utilize some Bootstrap
styling and add an icon to the button.

ryanahamilton committed:
Improve interaction with Recent Tasks/DAG Runs circles by ignoring pointer-events (#11786)

Resolves #11563. With the hover event triggering the tooltips, the tooltip was triggering a mouse-out event when the cursor was at the top of the circle (where the tooltip is presented). This was causing the fluttering behavior due to the looping off these events.

mjpieters committed:
Log task_instance execution duration as milliseconds (#10632)

This is best achieved by passing a `timedelta()` to `Stats.timing()`, and leave
worrying about time units to that method.

jbampton committed:
Standardize the Airflow CLI help descriptions (#11790)

Seltsamsel committed:
[AIRFLOW-6585] Fixed Timestamp bug in RefreshKubeConfigLoader (#11219)

Co-authored-by: Jan Brusch <jan.brusch@neuland-bfi.de>
Co-authored-by: Marian Cepok <marian.cepok@neuland-bfi.de>

ashb committed:
Fix locking issue stopping tasks running with SequentialExecutor (#11797)

Missing a commit means that the the row level lock was not released
before `executor.heartbeat()` was called.  This was only a problem for
the SequentialExecutor, as all the other executors would continue
running the scheduler code so the lock would be released shortly after
tasks are sent to the executor anyway. (Where as SequentialExecutor
doesn't return control until tasks have run!)

jedcunningham committed:
Fix backwards compatibility with k8s executor_config resources (#11796)

kaxil committed:
Add Project URLs for PyPI page (#11801)

These links would show up at https://pypi.org/project/apache-airflow/

kaxil committed:
Update download url for Airflow Version (#11800)

All the versions are available at https://archive.apache.org/dist/airflow/ and this link appears in https://pypi.org/project/apache-airflow/

kaxil committed:
Bump attrs to > 20.0 (#11799)

fixes https://github.com/apache/airflow/issues/11756

kaxil committed:
Remove loading DagBag multiple times in test_views.py:TestDecorators (#11804)

Previously we were loading DagBag twice, once in `setUpClass` and once in `setUp` (which is called everytime a test in that class is run)

kaxil committed:
Use Python 3 style super classes (#11806)

example:

```
super().__init__(label, validators, **kwargs)
```

instead of
```
super(DateTimeWithTimezoneField, self).__init__(label, validators, **kwargs)
```

kaxil committed:
Use PEP 380: Syntax for Delegating to a Subgenerator (#11805)

Since Master is Py 3.6>= we can use PEP 380 which was introduced in Python 3.3.

https://docs.python.org/3/whatsnew/3.3.html#pep-380

kaxil committed:
Remove redundant "UTF-8" in Python 3 (#11808)

```
In [1]: "result".encode("UTF-8")
Out[1]: b'result'

In [2]: "result".encode()
Out[2]: b'result'
```

kaxil committed:
Update to latest isort & pre-commit-hooks (#11813)

kaxil committed:
Remove redundant blank-line and parenthesis (#11811)

kaxil committed:
Replace non-empty sets with set literals (#11810)

Example:

```python
{paused_dag_id for paused_dag_id, in paused_dag_ids}
```

Instead of

```python
set(paused_dag_id for paused_dag_id, in paused_dag_ids)
```

kaxil committed:
Replace io.open with builtin open (#11807)



From Python 3 `io.open` is an alias for the builtin `open()` function.

Docs: https://docs.python.org/3/library/io.html#io.open

kaxil committed:
Use LocalExecutor by default with tmux + Breeze (#11791)

* Use LocalExecutor by default with tmux + Breeze

* Update run_tmux.sh

* Update run_tmux.sh

jbampton committed:
Fix spelling and grammar (#11814)

kaxil committed:
Remove redundant builtins imports (#11809)

These imports are not needed in Python3

potiuk committed:
Add links to PyPI packages. (#11818)

Same as in #11801 but in provider's packages.

jbampton committed:
Fix spelling (#11457)

jbampton committed:
Fix spelling (#11821)

potiuk committed:
Fixes flaky test test_should_response_200_with_reset_dag_run (#11817)

Sometimes the rows were returned in reverse order.

Fixes #11816

potiuk committed:
Switch postgres from 10 to 13 (#11785)

Seems that postgres is really stable when it comes to upgrades,
so we take the assumption that if we test 9.6 and 13, and they
work, all the versions between will also work.

This PR changes Postgres 10 to 13 in tests  and updates documentation
with all the versions in between.

kaxil committed:
Fix typo in scripts/in_container/entrypoint_ci.sh (#11824)

* Fix typo in scripts/in_container/entrypoint_ci.sh

* Update bats_tests.sh

kaxil committed:
Update download url for Airflow Providers Version (#11823)

All the versions are available at https://archive.apache.org/dist/airflow/ and this link appears in https://pypi.org/project/apache-airflow/

kaxil committed:
Fix bug when marking tasks when DAG Serialization is enabled (#11803)

This is because `current_task` is of SerializedBaseOperator type.

potiuk committed:
Removes duplicates from DISABLED_INTEGRATIONS variable (#11831)

Presto DB is checked several times but it also means that
it is added several times to DISABLED_INTEGRATIONS in case
it is not enabled. This commit fixes it.

potiuk committed:
Fixes typos in production-deployment docs (#11833)

mik-laj committed:
Fix example DAGs tests + add sanity checks (#11840)

potiuk committed:
Generated backport providers readmes/setup for 2020.10.29 (#11826)

ephraimbuddy committed:
Dag Run endpoints returns count total results after filtering (#11832)

ephraimbuddy committed:
Dag Run endpoints returns count total results after filtering (#11832)

jbampton committed:
Fix spellings (#11825)

jbampton committed:
Fix spellings (#11825)

potiuk committed:
Fix the script that builds source for backports (#11846)

First time preparing backports after converting scripts to
also support regular providers. Some small bugs were found
and fixed.

potiuk committed:
Fix the script that builds source for backports (#11846)

First time preparing backports after converting scripts to
also support regular providers. Some small bugs were found
and fixed.

yangrong688 committed:
Fix commands in docs/usage-cli.rst(#11847)

* Add uSmart Securities to the INTHEWILD.md

* change airflow dag to airflow dags in 2.0

yangrong688 committed:
Fix commands in docs/usage-cli.rst(#11847)

* Add uSmart Securities to the INTHEWILD.md

* change airflow dag to airflow dags in 2.0

potiuk committed:
Few more small fixes found during preparation of the backports (#11848)

potiuk committed:
Few more small fixes found during preparation of the backports (#11848)

potiuk committed:
Local Executor is used by default for MySQL/Postgres breeze (#11792)

potiuk committed:
Local Executor is used by default for MySQL/Postgres breeze (#11792)

Commit:
Google Memcached hooks - improve protobuf messages handling (#11743)

Commit:
Google Memcached hooks - improve protobuf messages handling (#11743)

mpermana committed:
Unauthenticated access with RBAC to URL has_dag_access results lose redirection (#11592)

Co-authored-by: Michael Permana <mpermana@pinterest.com>

Closes #11591

mpermana committed:
Unauthenticated access with RBAC to URL has_dag_access results lose redirection (#11592)

Co-authored-by: Michael Permana <mpermana@pinterest.com>

Closes #11591

potiuk committed:
The "OpenAPI" tests missed dependency on build-info step (#11849)

The tests were not running for a few days because the
dependency (needs) was missing - it was needed to get the
selective checks work after the recent refactor in #11656.

potiuk committed:
The "OpenAPI" tests missed dependency on build-info step (#11849)

The tests were not running for a few days because the
dependency (needs) was missing - it was needed to get the
selective checks work after the recent refactor in #11656.

ashb committed:
Use packaging.version, not semver module for version comparisons (#11854)

Semver module doesn't like python version specifiers such as `0.0.2a1`
-- since packaging module is already a dep from setup tools, and is what
the python ecosystem uses to do version handling it makes sense to use
it.

ashb committed:
Use packaging.version, not semver module for version comparisons (#11854)

Semver module doesn't like python version specifiers such as `0.0.2a1`
-- since packaging module is already a dep from setup tools, and is what
the python ecosystem uses to do version handling it makes sense to use
it.

ashb committed:
Prepare providers release 0.0.2a1 (#11855)

ashb committed:
Prepare providers release 0.0.2a1 (#11855)

casassg committed:
Add @dag decorator (#10587)

casassg committed:
Add @dag decorator (#10587)

potiuk committed:
Occasional docker-compose errors will be easier to diagnose (#11835)

With this change we attempt to better diagnose some occasional
network docker-compose issues that have beeen plaguing us after
we solved or workarounded other CI-related issues. Sometimes
the docker compose jobs fail on checking if the container is
up and running with either of the two errors:

 * 'forward host lookup failed: Unknown host`
 * 'DNS fwd/rev mismatch'

Usually this happens in rabbitMQ and openldap containers.

Both indicate a problem with DNS of the docker engine or maybe
some remnants of the previous docker run that do not allow us
to start those containers.

This change introduces few improvements:

* added --volume in `docker system prune` command which might
  clean-up some anonymous volumes left by the containers between
  runs

* removed docker-compose down --remove-orphans --down command
  after failure, as currently we are anyhow always doing it
  few lines before (before the test). This change will cause
  that our mechanism of logging container logs after failure
  will likely give us more information about in case the root
  cause is rabbitmq or openldap container failing to start

* Increases number of tries to 5 in case of failed containers.

potiuk committed:
Occasional docker-compose errors will be easier to diagnose (#11835)

With this change we attempt to better diagnose some occasional
network docker-compose issues that have beeen plaguing us after
we solved or workarounded other CI-related issues. Sometimes
the docker compose jobs fail on checking if the container is
up and running with either of the two errors:

 * 'forward host lookup failed: Unknown host`
 * 'DNS fwd/rev mismatch'

Usually this happens in rabbitMQ and openldap containers.

Both indicate a problem with DNS of the docker engine or maybe
some remnants of the previous docker run that do not allow us
to start those containers.

This change introduces few improvements:

* added --volume in `docker system prune` command which might
  clean-up some anonymous volumes left by the containers between
  runs

* removed docker-compose down --remove-orphans --down command
  after failure, as currently we are anyhow always doing it
  few lines before (before the test). This change will cause
  that our mechanism of logging container logs after failure
  will likely give us more information about in case the root
  cause is rabbitmq or openldap container failing to start

* Increases number of tries to 5 in case of failed containers.

ashb committed:
Cut 2.0.0alpha2 (#11860)

ashb committed:
Cut 2.0.0alpha2 (#11860)

kaxil committed:
Update instruction to only push a single tag while releasing Airflow (#11861)

kaxil committed:
Update instruction to only push a single tag while releasing Airflow (#11861)

kaxil committed:
Make Dag Serialization a hard requirement (#11335)

Scheduler HA uses Serialized DAGs and hence it is a strict
the requirement for 2.0.

It also has performance benefits for the Webserver and so should
be used by default anyway.

Task execution on workers will continue to use the actual files for execution.

Scheduler, Experimental API and Webserver will read the DAGs from DB using
`DagBag(read_dags_from_db=True)`

kaxil committed:
Make Dag Serialization a hard requirement (#11335)

Scheduler HA uses Serialized DAGs and hence it is a strict
the requirement for 2.0.

It also has performance benefits for the Webserver and so should
be used by default anyway.

Task execution on workers will continue to use the actual files for execution.

Scheduler, Experimental API and Webserver will read the DAGs from DB using
`DagBag(read_dags_from_db=True)`

michael-j-thomas committed:
Add FanDuel to the list of users in the wild.  (#11864)

michael-j-thomas committed:
Add FanDuel to the list of users in the wild.  (#11864)

kaxil committed:
Pin werkzeug back to <1.0.0 (#11872)

closes https://github.com/apache/airflow/issues/11871

kaxil committed:
Pin werkzeug back to <1.0.0 (#11872)

closes https://github.com/apache/airflow/issues/11871

kaxil committed:
Fix failing tests in tests/models/test_dag.py (#11868)

kaxil committed:
Fix failing tests in tests/models/test_dag.py (#11868)

kaxil committed:
Use correct name for PostgreSQL (#11869)

`PostgresSQL` -> `PostgreSQL`

kaxil committed:
Use correct name for PostgreSQL (#11869)

`PostgresSQL` -> `PostgreSQL`

eladkal committed:
add wix & wixanswers (#11863)

eladkal committed:
add wix & wixanswers (#11863)

potiuk committed:
Retrieve PR labels from the associated PR (#11820)

This change is needed to implement optimisation of running only
limited version of the matrix tests for PRs that are not yet
approved by committers.

potiuk committed:
Retrieve PR labels from the associated PR (#11820)

This change is needed to implement optimisation of running only
limited version of the matrix tests for PRs that are not yet
approved by committers.

potiuk committed:
Constraints generation runs regardless from test status (#11838)

Constraints generation did not run when tests were failing
but this was wrong. In case of tests that fail due to constraints
upgrade we neeed to see the output of constraint generation, to
know which constraints failed. Only pushing the constraint
should be limited to the case where everything succeeded.

This should make investigation of problems similar to #11837 much
easier.

potiuk committed:
Constraints generation runs regardless from test status (#11838)

Constraints generation did not run when tests were failing
but this was wrong. In case of tests that fail due to constraints
upgrade we neeed to see the output of constraint generation, to
know which constraints failed. Only pushing the constraint
should be limited to the case where everything succeeded.

This should make investigation of problems similar to #11837 much
easier.

potiuk committed:
Adds newly added fast CI jobs to "fail-fast" cancelling (#11830)

We "fail-fast" the whole workflow run if we the queue is busy and
we notice that there is a failing "fast" job - i.e. job that
probably failed rather fast, and you will have to rerun the build
anyway, so cancelling tests for the build that is going to be
re-pushed again is a nice thing to do for others.

There were two jobs added recently that did not match the
regexp for job names.

potiuk committed:
Adds newly added fast CI jobs to "fail-fast" cancelling (#11830)

We "fail-fast" the whole workflow run if we the queue is busy and
we notice that there is a failing "fast" job - i.e. job that
probably failed rather fast, and you will have to rerun the build
anyway, so cancelling tests for the build that is going to be
re-pushed again is a nice thing to do for others.

There were two jobs added recently that did not match the
regexp for job names.

potiuk committed:
Fixes the doc pattern in selective checks (#11834)

The pattern contained $ which effectively stopped docs from
being run on doc-only change :(

potiuk committed:
Fixes the doc pattern in selective checks (#11834)

The pattern contained $ which effectively stopped docs from
being run on doc-only change :(

kaxil committed:
Upgrade FAB to 3.1.1 (#11884)

We can also remove the FAB Actions fix with composite PKs (https://github.com/apache/airflow/pull/11753) since it is merged and release in FAB 3.1.1

kaxil committed:
Upgrade FAB to 3.1.1 (#11884)

We can also remove the FAB Actions fix with composite PKs (https://github.com/apache/airflow/pull/11753) since it is merged and release in FAB 3.1.1

ryanahamilton committed:
Revert "fix: Override FAB styling (#11752)" (#11888)

This reverts commit 2df9d1c4d78d0add5efa18831725ddcc2a93d738.

ryanahamilton committed:
Revert "fix: Override FAB styling (#11752)" (#11888)

This reverts commit 2df9d1c4d78d0add5efa18831725ddcc2a93d738.

anirudhbagri committed:
Fix broken link in README.md (#11885)

* Link to committers list was broken

anirudhbagri committed:
Fix broken link in README.md (#11885)

* Link to committers list was broken

anirudhbagri committed:
Update start.rst (#11886)

anirudhbagri committed:
Update start.rst (#11886)

potiuk committed:
Fixes a problem with checked-out version of the selective check (#11891)

The non-master version of selective check script has
been checked out by mistake in the build-info step.

potiuk committed:
Fixes a problem with checked-out version of the selective check (#11891)

The non-master version of selective check script has
been checked out by mistake in the build-info step.

potiuk committed:
Fixes backtick in the documentation (#11892)

potiuk committed:
Fixes backtick in the documentation (#11892)

petedejoy committed:
Add providers, operators, and hooks readmes (#11829)

* Add readmes with relevant information for Airflow 2.0 to providers, operators, and hooks directories

* Add newline to end of files

* Remove hanging EOF empty line

petedejoy committed:
Add providers, operators, and hooks readmes (#11829)

* Add readmes with relevant information for Airflow 2.0 to providers, operators, and hooks directories

* Add newline to end of files

* Remove hanging EOF empty line

kaxil committed:
Fix typo in the word daemon (#11897)

`deamon` -> `daemon`

kaxil committed:
Fix typo in the word daemon (#11897)

`deamon` -> `daemon`

dimberman committed:
Updating 2.0 docs (#11842)

* Separate 2.0 upgrade steps into their own file

* shell script

* more changes

dimberman committed:
Updating 2.0 docs (#11842)

* Separate 2.0 upgrade steps into their own file

* shell script

* more changes

petedejoy committed:
Add readme for core sensors, standardize capitalization schema (#11898)

petedejoy committed:
Add readme for core sensors, standardize capitalization schema (#11898)

dimberman committed:
Change Decorated Flow to TaskFlow API (#11895)

* Change Decorated Flow to TaskFlow API

In order to better describe what this new data transfer system does,
we are changing the name of Decorated Flows to TaskFlow API

* Update docs/concepts.rst

Co-authored-by: Gerard Casas Saez <casassg@users.noreply.github.com>

* Update docs/concepts.rst

Co-authored-by: Kaxil Naik <kaxilnaik@gmail.com>

* Update docs/tutorial_taskflow_api.rst

Co-authored-by: Kaxil Naik <kaxilnaik@gmail.com>

Co-authored-by: Gerard Casas Saez <casassg@users.noreply.github.com>
Co-authored-by: Kaxil Naik <kaxilnaik@gmail.com>

dimberman committed:
Change Decorated Flow to TaskFlow API (#11895)

* Change Decorated Flow to TaskFlow API

In order to better describe what this new data transfer system does,
we are changing the name of Decorated Flows to TaskFlow API

* Update docs/concepts.rst

Co-authored-by: Gerard Casas Saez <casassg@users.noreply.github.com>

* Update docs/concepts.rst

Co-authored-by: Kaxil Naik <kaxilnaik@gmail.com>

* Update docs/tutorial_taskflow_api.rst

Co-authored-by: Kaxil Naik <kaxilnaik@gmail.com>

Co-authored-by: Gerard Casas Saez <casassg@users.noreply.github.com>
Co-authored-by: Kaxil Naik <kaxilnaik@gmail.com>

kaxil committed:
Unpin werkzeug & set default cookie_samesite to Lax (#11873)

As suggested by https://flask.palletsprojects.com/en/1.1.x/config/#SESSION_COOKIE_SAMESITE , set `cookie_samesite` to `Lax`.

This Werkzeug bug prevented using the latest version of Werkzeug (1.1.0): https://github.com/pallets/werkzeug/issues/1549 | https://github.com/pallets/werkzeug/pull/1550

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie
Detailed explanation: https://web.dev/samesite-cookies-explained/

kaxil committed:
Unpin werkzeug & set default cookie_samesite to Lax (#11873)

As suggested by https://flask.palletsprojects.com/en/1.1.x/config/#SESSION_COOKIE_SAMESITE , set `cookie_samesite` to `Lax`.

This Werkzeug bug prevented using the latest version of Werkzeug (1.1.0): https://github.com/pallets/werkzeug/issues/1549 | https://github.com/pallets/werkzeug/pull/1550

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie
Detailed explanation: https://web.dev/samesite-cookies-explained/

kaxil committed:
Fix broken docs build (#11900)

kaxil committed:
Fix broken docs build (#11900)

dimberman committed:
Add Python Helm testing framework (#11693)

* Helm Python Testing

* helm change

* add back args

dimberman committed:
Add Python Helm testing framework (#11693)

* Helm Python Testing

* helm change

* add back args

dimberman committed:
Add taskflow to accepted words (#11902)

dimberman committed:
Add taskflow to accepted words (#11902)

JavierLopezT committed:
Add truncate table (before copy) option to S3ToRedshiftOperator (#9246)

- add table arg to jinja template fields
- change ui_color

Co-authored-by: javier.lopez <javier.lopez@promocionesfarma.com>

JavierLopezT committed:
Add truncate table (before copy) option to S3ToRedshiftOperator (#9246)

- add table arg to jinja template fields
- change ui_color

Co-authored-by: javier.lopez <javier.lopez@promocionesfarma.com>

kaxil committed:
Fix CI Step Name for Postgres (#11908)

https://github.com/apache/airflow/commit/0d1ad6648ee63311614e043d5893cf36a0cd9aea updated the Step name by Mistake from `"Tests: ${{needs.build-info.outputs.testTypes}}"` to `"Tests: Helm"`

kaxil committed:
Fix CI Step Name for Postgres (#11908)

https://github.com/apache/airflow/commit/0d1ad6648ee63311614e043d5893cf36a0cd9aea updated the Step name by Mistake from `"Tests: ${{needs.build-info.outputs.testTypes}}"` to `"Tests: Helm"`

potiuk committed:
Constraints job depends on CI images (#11904)

The constraints dependencies removed in #11838
removed ci-images implicit dependency which is needed for the
constraints step to complete. This caused occasional failures in
the push /merges builds.

potiuk committed:
Constraints job depends on CI images (#11904)

The constraints dependencies removed in #11838
removed ci-images implicit dependency which is needed for the
constraints step to complete. This caused occasional failures in
the push /merges builds.

kaxil committed:
Fix Helm Chart Testing guide (#11909)

kaxil committed:
Fix Helm Chart Testing guide (#11909)

Commit:
Add note on GKE metadata server Workload Identity (#10728)

* Add note on GKE metadata server Workload Identity

cc: @mik-laj @potiuk

* fixup! move detailed notes on ADC to separate section

* fixup! remove trailing whitespace

Commit:
Add note on GKE metadata server Workload Identity (#10728)

* Add note on GKE metadata server Workload Identity

cc: @mik-laj @potiuk

* fixup! move detailed notes on ADC to separate section

* fixup! remove trailing whitespace

kaxil committed:
Fix broken doc build on Master (#11915)

kaxil committed:
Fix broken doc build on Master (#11915)

nathadfield committed:
Fixing re pattern and changing to use a single character class. (#11857)

nathadfield committed:
Fixing re pattern and changing to use a single character class. (#11857)

kaxil committed:
Add Ryan Hamilton to Committers list (#11923)

kaxil committed:
Add Ryan Hamilton to Committers list (#11923)

ephraimbuddy committed:
Make taskinstances pid and duration nullable (#11906)

ephraimbuddy committed:
Make taskinstances pid and duration nullable (#11906)

potiuk committed:
Simplify "scheduled" conditons to follow today's change in GA (#11876)

GitHub Actions policy about runnig scheduled workflows in forks
has changed recently and as of October 27 2020 scheduled workflows
in forks will be disabled. Thanks to that change we can simplify
some of our conditions that disallowed running scheduled workflows
in forks.

The change is active as of today (email from GitHub):

> What will happen to scheduled workflows in forks I already have?

> If you already have scheduled workflows in forks of public
  repositories, they will be disabled on October 27, 2020. Repository
  owners will receive a reminder email 7 days prior and can choose to
  keep the workflows running at that time if they are needed.

potiuk committed:
Simplify "scheduled" conditons to follow today's change in GA (#11876)

GitHub Actions policy about runnig scheduled workflows in forks
has changed recently and as of October 27 2020 scheduled workflows
in forks will be disabled. Thanks to that change we can simplify
some of our conditions that disallowed running scheduled workflows
in forks.

The change is active as of today (email from GitHub):

> What will happen to scheduled workflows in forks I already have?

> If you already have scheduled workflows in forks of public
  repositories, they will be disabled on October 27, 2020. Repository
  owners will receive a reminder email 7 days prior and can choose to
  keep the workflows running at that time if they are needed.

potiuk committed:
Fix spelling problem introduced in #11923 (#11927)

potiuk committed:
Fix spelling problem introduced in #11923 (#11927)

ashb committed:
Add docs about Scheduler HA, how to use it and DB requirements (#11467)

ashb committed:
Add docs about Scheduler HA, how to use it and DB requirements (#11467)

jhtimmins committed:
Use resource permissions for Airflow view access (#11362)

Migrate the Airflow view to use resource-backed permissions. This includes creation of a custom has_access decorator, deprecation of the has_dag_accessdecorator, a migration script to update existing roles to use the new permissions, and adding the new has_access decorator to the Airflow view class's methods.

jhtimmins committed:
Use resource permissions for Airflow view access (#11362)

Migrate the Airflow view to use resource-backed permissions. This includes creation of a custom has_access decorator, deprecation of the has_dag_accessdecorator, a migration script to update existing roles to use the new permissions, and adding the new has_access decorator to the Airflow view class's methods.

potiuk committed:
The PRs which are not approved run subset of tests (#11828)

This PR is an implementation of optimisation - to only run
default values for build matrix in case PR does not have
"okay to test" label.

This "okay to test" label is set when the PR gets approved
but it was not approved before, also then a comment is generated
urging the committer to rebase the PR to run full set of tests.

Additionally a check is added (in-progress) that makes the PR
not yet ready to be merged. Only after re-running it it will
become truly readty to be merged.

potiuk committed:
The PRs which are not approved run subset of tests (#11828)

This PR is an implementation of optimisation - to only run
default values for build matrix in case PR does not have
"okay to test" label.

This "okay to test" label is set when the PR gets approved
but it was not approved before, also then a comment is generated
urging the committer to rebase the PR to run full set of tests.

Additionally a check is added (in-progress) that makes the PR
not yet ready to be merged. Only after re-running it it will
become truly readty to be merged.

potiuk committed:
Quarantine test_exception_propagation (#11933)

Issue created #11932

potiuk committed:
Quarantine test_exception_propagation (#11933)

Issue created #11932

juanuicich committed:
Add Umami Collective as Airflow user (#11935)

juanuicich committed:
Add Umami Collective as Airflow user (#11935)

kaxil committed:
ci: Fix CodeQL Workflow for Javascript (#11941)

We removed a bracket in bdd5e0b#diff-63bd641104d10e25f141d518a16b22a151d125e12701df2f9e79734b23b90188 due to which one of the conditions for CodeQL javascript has changed by mistake, so it is trying to run git checkout HEAD^2 for merged commits too instead of just running it on PRs.

Example failed run on Master: https://github.com/apache/airflow/runs/1326054227

kaxil committed:
ci: Fix CodeQL Workflow for Javascript (#11941)

We removed a bracket in bdd5e0b#diff-63bd641104d10e25f141d518a16b22a151d125e12701df2f9e79734b23b90188 due to which one of the conditions for CodeQL javascript has changed by mistake, so it is trying to run git checkout HEAD^2 for merged commits too instead of just running it on PRs.

Example failed run on Master: https://github.com/apache/airflow/runs/1326054227

ashb committed:
Update flask_wtf version to work with werkzeug>=1.0 (#11939)

Werkzeug 0.16 deprecated werkzeug.url_encode, and removed it in 1.0, so
we need the fixed version of flask_wtf

ashb committed:
Update flask_wtf version to work with werkzeug>=1.0 (#11939)

Werkzeug 0.16 deprecated werkzeug.url_encode, and removed it in 1.0, so
we need the fixed version of flask_wtf

kaxil committed:
Use Github Discussions to asking User Questions (#11940)

kaxil committed:
Use Github Discussions to asking User Questions (#11940)

Commit:
Add drain option when canceling Dataflow pipelines (#11374)

* Add drain option when cancel Dataflow pipelines

* fixup! Add drain option when cancel Dataflow pipelines

* fixup! fixup! Add drain option when cancel Dataflow pipelines

* fixup! fixup! fixup! Add drain option when cancel Dataflow pipelines

Commit:
Add drain option when canceling Dataflow pipelines (#11374)

* Add drain option when cancel Dataflow pipelines

* fixup! Add drain option when cancel Dataflow pipelines

* fixup! fixup! Add drain option when cancel Dataflow pipelines

* fixup! fixup! fixup! Add drain option when cancel Dataflow pipelines

jbampton committed:
Standardize quotes in HTML files (#11724)

jbampton committed:
Standardize quotes in HTML files (#11724)

rynmccrmck committed:
Add Template Fields to RedshiftToS3Operator & S3ToRedshiftOperator (#11844)

rynmccrmck committed:
Add Template Fields to RedshiftToS3Operator & S3ToRedshiftOperator (#11844)

francescomucio committed:
Add How-to guide for JDBC Operator (#11472)

francescomucio committed:
Add How-to guide for JDBC Operator (#11472)

kaxil committed:
Remove archived link from README.md (#11945)

closes https://github.com/apache/airflow/issues/11943

kaxil committed:
Remove archived link from README.md (#11945)

closes https://github.com/apache/airflow/issues/11943

ashb committed:
Rename example JDBC dag (#11946)

This was merged in #11472, but for some reason that didn't run tests so
the failure wasn't noticed

ashb committed:
Rename example JDBC dag (#11946)

This was merged in #11472, but for some reason that didn't run tests so
the failure wasn't noticed

kaxil committed:
Fix failing docs build on Master (#11951)

This was merged in #11472, but for some reason that didn't run tests so
the failure wasn't noticed.

a7ad204 fixed 1 error but the docs error was missed

kaxil committed:
Fix failing docs build on Master (#11951)

This was merged in #11472, but for some reason that didn't run tests so
the failure wasn't noticed.

a7ad204 fixed 1 error but the docs error was missed

dimberman committed:
Fix issue rendering k8s V1Pod (#11952)

* Fix issue rendering k8s V1Pod

Adds return statement for json serialization of k8s.V1Pod that
previously caused errors in UI

* fix task_instances

dimberman committed:
Fix issue rendering k8s V1Pod (#11952)

* Fix issue rendering k8s V1Pod

Adds return statement for json serialization of k8s.V1Pod that
previously caused errors in UI

* fix task_instances

kaxil committed:
Fix broken master (isort fix) (#11954)

Static checks are failing because of a Bad merge to Master.

kaxil committed:
Fix broken master (isort fix) (#11954)

Static checks are failing because of a Bad merge to Master.

ryanahamilton committed:
Fix oversized width of DAGs table with hide/reveal of "links" (#11866)

* Conserve horizontal space by adding hide/reveal "links" in DAGs table

* Reverse the order of links to reduce mouse distance for most popular

ryanahamilton committed:
Fix oversized width of DAGs table with hide/reveal of "links" (#11866)

* Conserve horizontal space by adding hide/reveal "links" in DAGs table

* Reverse the order of links to reduce mouse distance for most popular

ashb committed:
Make mypy happy with airflow.executors.local_executor (#11944)

This is an odd one -- in making a change in another file, this started
erroring - I guess it gets confused about the fork, but I'm not sure
what changed to cause this to become a problem.

ashb committed:
Make mypy happy with airflow.executors.local_executor (#11944)

This is an odd one -- in making a change in another file, this started
erroring - I guess it gets confused about the fork, but I'm not sure
what changed to cause this to become a problem.

ashb committed:
Add missing space to log message in task completion logs (#11934)

Previously it looked like this

    INFO - Marking task as SUCCESS.dag_id=scenario1_case1_1_1, ...

Now it looks like

    INFO - Marking task as SUCCESS. dag_id=scenario1_case1_1_1, ...

ashb committed:
Add missing space to log message in task completion logs (#11934)

Previously it looked like this

    INFO - Marking task as SUCCESS.dag_id=scenario1_case1_1_1, ...

Now it looks like

    INFO - Marking task as SUCCESS. dag_id=scenario1_case1_1_1, ...

ashb committed:
Cache CLI parser objects (#11957)

Right now, when we are using the CLI parser in LocalTaskJob and
executor, having to re-create the parser each time adds a noticeable
overhead for quick tasks -- by caching this we save 0.07s

(I know that doesn't sound like much, but for an `echo true`
BashOperator task that takes it from 0.25s to 0.19s - i.e. it cuts the
time down by about 20%!)

ashb committed:
Cache CLI parser objects (#11957)

Right now, when we are using the CLI parser in LocalTaskJob and
executor, having to re-create the parser each time adds a noticeable
overhead for quick tasks -- by caching this we save 0.07s

(I know that doesn't sound like much, but for an `echo true`
BashOperator task that takes it from 0.25s to 0.19s - i.e. it cuts the
time down by about 20%!)

ashb committed:
Update flask-caching dep to version that works with Werkzeug 1 (#11955)

Werkzeug 1.0 removed the werkzeug.contrib.caching module, so we need at
least 1.5.0 of flask-caching to deal with this -- see

https://github.com/sh4nks/flask-caching/blob/v1.9.0/CHANGES#L97-L107

ashb committed:
Update flask-caching dep to version that works with Werkzeug 1 (#11955)

Werkzeug 1.0 removed the werkzeug.contrib.caching module, so we need at
least 1.5.0 of flask-caching to deal with this -- see

https://github.com/sh4nks/flask-caching/blob/v1.9.0/CHANGES#L97-L107

ashb committed:
Speed up task execution in Celery by pre-loading "expensive" modules (#11956)

These modules add about 0.3s to _every_ task run time, which is nothing
for a long task, but significant if you want to run a lot of short tasks
very quickly.

These modules were discovered by running with PYTHONPROFILEIMPORTTIME=1
environment variable set and looking at what was imported and expensive.

ashb committed:
Speed up task execution in Celery by pre-loading "expensive" modules (#11956)

These modules add about 0.3s to _every_ task run time, which is nothing
for a long task, but significant if you want to run a lot of short tasks
very quickly.

These modules were discovered by running with PYTHONPROFILEIMPORTTIME=1
environment variable set and looking at what was imported and expensive.

jhtimmins committed:
Migration commands shouldn't print a bunch of data. (#11961)

jhtimmins committed:
Migration commands shouldn't print a bunch of data. (#11961)

kaxil committed:
Log instead of raise an Error for unregistered OperatorLinks (#11959)

Currently, if someone uses OperatorLinks that are not registered,
it will break the UI when someone clicks on that DAG.

This commit will instead log an error in the Webserver logs so that
someone can still see the DAG in different Views (graph, tree, etc).

kaxil committed:
Log instead of raise an Error for unregistered OperatorLinks (#11959)

Currently, if someone uses OperatorLinks that are not registered,
it will break the UI when someone clicks on that DAG.

This commit will instead log an error in the Webserver logs so that
someone can still see the DAG in different Views (graph, tree, etc).

mik-laj commented on airflow/www/app.py:
```diff
@@ -66,8 +66,19 @@ def create_app(config=None, testing=False, app_name="Airflow"):
+        session_lifetime_minutes = conf.getint('webserver', 'SESSION_LIFETIME_MINUTES',
+                                               fallback=default_session_lifetime_minutes)
+    else:
+        warnings.warn('`SESSION_LIFETIME_DAYS` option has been renamed to `SESSION_LIFETIME_MINUTES`. '
```
Can you also add the section name to this warning message?

michalmisiewicz replied:
done

mik-laj reviewed (approved):
Small nit.

michalmisiewicz committed:
add information about section in deprecation warning

michalmisiewicz committed:
add information about section in deprecation warning

mjpieters committed:
Clean up command-line arguments (#11682)

* Clean up command-line arguments

- The `--do-pickle` argument is exclusively used by the scheduler, the `worker` command completely ignores it.
- Remove the `ARG_DAG_ID_OPT` entry entirely. The scheduler doesn't take a `--dag-id` option, and while the `dags list-runs` and `dags list-jobs` commands *do* take such a switch, the help text here is incorrect, they simply need to use `ARG_DAG_ID`.

* Put ARG_DAG_ID_OPT back

It differs from ARG_DAG_ID in that it is an optional command-line switch. I changed the help text to match its use: to select a specific DAG when listing runs or jobs. The dag is not actually triggered.

mjpieters committed:
Clean up command-line arguments (#11682)

* Clean up command-line arguments

- The `--do-pickle` argument is exclusively used by the scheduler, the `worker` command completely ignores it.
- Remove the `ARG_DAG_ID_OPT` entry entirely. The scheduler doesn't take a `--dag-id` option, and while the `dags list-runs` and `dags list-jobs` commands *do* take such a switch, the help text here is incorrect, they simply need to use `ARG_DAG_ID`.

* Put ARG_DAG_ID_OPT back

It differs from ARG_DAG_ID in that it is an optional command-line switch. I changed the help text to match its use: to select a specific DAG when listing runs or jobs. The dag is not actually triggered.

potiuk committed:
Move test_serve_logs to quarantine as it fails on mysql sometimes (#11963)

Issue recorded at #11962

potiuk committed:
Move test_serve_logs to quarantine as it fails on mysql sometimes (#11963)

Issue recorded at #11962

potiuk committed:
Moves tests that should be always executed to 'always' directory (#11948)

Some tests (testing the structure and importability of
example) should be always run even if core part was not modified.

That's why we move it to "always" directory.

potiuk committed:
Moves tests that should be always executed to 'always' directory (#11948)

Some tests (testing the structure and importability of
example) should be always run even if core part was not modified.

That's why we move it to "always" directory.

stijn-meersman committed:
Add autocommit property for snowflake connection (#10838)

stijn-meersman committed:
Add autocommit property for snowflake connection (#10838)

ryanahamilton committed:
Fix: Responsive layout of DAGs (Home) view (#11958)

ryanahamilton committed:
Fix: Responsive layout of DAGs (Home) view (#11958)

michalmisiewicz committed:
Merge remote-tracking branch 'origin/unify-user-session-lifetime' into unify-user-session-lifetime
